src/Http/Authentication.Abstractions/src/IAuthenticationSchemeProvider.cs

```diff
@@ -71,6 +71,13 @@ public interface IAuthenticationSchemeProvider
         /// <param name="scheme">The scheme.</param>
         void AddScheme(AuthenticationScheme scheme);
 
+        /// <summary>
+        /// Registers a scheme for use by <see cref="IAuthenticationService"/>. 
+        /// </summary>
+        /// <param name="scheme">The scheme.</param>
+        /// <returns>true if the scheme was added successfully.</returns>
+        bool TryAddScheme(AuthenticationScheme scheme) => false;
```

Comment:
You didn't want to call AddScheme and return true? It works fine most of the time.

Reply:
That depends on what the contract of this method is. What kind of "Try" is it? Is it "try and if there's an exception, return false instead" or is it "add it if it hasn't already been added"?

Reply:
From the [original issue](https://github.com/aspnet/AspNetCore/issues/4666):

> Instead perhaps this method should be bool TryAddScheme(AuthenticationScheme scheme) so users don't have to wrap it in a try-catch.

Maybe the default implementation should be `try { AddScheme(scheme); return true } catch { return false; }` ?

Reply:
Yeah try catch is the correct behavior, this is mostly to prevent exceptions